code_generator_helpers/module.py

```diff
@@ -475,26 +474,26 @@ def union(self, union, name):
                         assert len(parts) == 1
                         self.out("Ok(%s)", parts[0])
                     self.out("}")
-                    self.out("do_the_parse(&self.0[..]).unwrap()")
+                    self.out("do_the_parse(&self.0).unwrap()")
                 self.out("}")
         self.out("}")
 
         self.out("impl Serialize for %s {", rust_name)
         with Indent(self.out):
-            self.out("type Bytes = Vec<u8>;")
+            self.out("type Bytes = [u8; %s];", union_size)
             self.out("fn serialize(&self) -> Self::Bytes {")
-            self.out.indent("self.0.clone()")
+            self.out.indent("self.0")
             self.out("}")
         self.out("}")
 
         self.out("impl TryParse for %s {", rust_name)
         with Indent(self.out):
             self.out("fn try_parse(value: &[u8]) -> Result<(Self, &[u8]), ParseError> {")
             with Indent(self.out):
-                union_size = union.get_total_size()
-                self.out("let inner = value.get(..%s)", union_size)
+                self.out("let inner: [u8; %s] = value.get(..%s)", union_size, union_size)
                 self.out.indent(".ok_or(ParseError::ParseError)?")
-                self.out.indent(".iter().copied().collect();")
+                self.out.indent(".try_into()")
```

Comment:
This uses `impl<T> TryFrom<&[T]> for [T; N]`, right? As such, this will only work with unions which have a length of at most 32. Since this seems to be the case with all unions: Okay, fine with me. Just wanted to make sure.

Reply:
Indeed. Hopefully, nobody will extend a X11 extension with a larger union soon.

Reply:
Hopefully, nobody will extend an X11 extension with a union. Ever. :-)